repair/row_level.cc

```diff
@@ -767,6 +774,10 @@ class repair_meta {
     bool _small_table_optimization_tm_calculated = false;
     service::frozen_topology_guard _frozen_topology_guard;
     service::topology_guard _topology_guard;
+    // Max fragments size and number are allowed in a single repair_row_on_wire
+    // which is sent as a rpc stream message
+    static constexpr size_t _max_fragments_size = 240 * 1024;
+    static constexpr size_t _max_fragments_nr = 32 * 1024;
```

Comment:
But only if REPAIR_MSG_SPLIT is negotiated.

Note: REPAIR_MSG_SPLIT could appear in the middle of a repair. It's not correct for server parts of a client/server system to rely on it.

The correct approach is for the RPC that initiates the repair (the root of the RPC tree) to copy REPAIR_MSG_SPLIT into a request parameter, and everything else copies the flag from there, so everyone has a consistent view.

Reply:
Please add documentation to idl/ to explain how it behaves differently. Usually we have [[version blah]] to indicate what changed (would be better to have [[feature blah]] now that I think about it).

I'd even prefer a new flag in repair_rows_on_wire than subtle changes in meaning of existing fields.

Reply:
We still need the gossip feature bit otherwise the receiver with older version could not handle the split message.  I think it is better to add a new flag instead of using empty partition_key to ask the receiver to use the previous partition key. 

I will send a new version.